mpqp/noise/noise_model.py

```diff
@@ -454,7 +452,7 @@ def __init__(
     def to_kraus_representation(self) -> KrausRepresentation: ...
 
     def __repr__(self):
-        targets = f", targets={self.targets}" if len(self.targets) != 0 else ""
+        targets = f", targets={self.targets}"
         gates = f", gates={self.gates}" if len(self.gates) != 0 else ""
         prob = f", prob={self.prob}" if self.prob != 1 else ""
```

Comment:
if prob is not equal to 1, then empty string will be put after self.gamma in the __repr__, so we will display directly the third argument (targets), while we don't use keywords. I think this will result in a wrong repr.

Example : 
If I instantiate the object `a = AmplitudeDamping(0.3, targets=[0,1])`, the output repr will be `AmplitudeDamping(0.3, [0,1])`, which is a wrong __repr__.

Reply:
it will print  AmplitudeDamping(0.3, targets=[0,1]):
targets = f", targets={self.targets}"                                      ->  targets=[0,1]
gates = f", gates={self.gates}" if len(self.gates) != 0 else "" -> ""
prob = f", prob={self.prob}" if self.prob != 1 else ""            -> ""
return f"{type(self).__name__}({self.gamma}{prob}{targets}{gates})" -> AmplitudeDamping({0.3}{""}  {, targets=[0,1]} {"}):
                                                                                                                                                    {prob}     {targets}    {gates}

Reply:
why did you remove this targets length check ? is seems useful to me
I don't think Hamza is right in his example, but I still think that for an empty target, you should not have in in the repr, were am I wrong ?

Reply:
true i discard the change. I think it was to specify the target for all the qubits, but we have to do it in __str__.

Reply:
Indeed my example was not right.